core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -481,8 +481,14 @@ object SparkSubmit {
     }
 
     // Let YARN know it's a pyspark app, so it distributes needed libraries.
-    if (clusterManager == YARN && args.isPython) {
-      sysProps.put("spark.yarn.isPython", "true")
+    if (clusterManager == YARN) {
+      if (args.isPython) {
+        sysProps.put("spark.yarn.isPython", "true")
+      }
+      if (args.principal != null) {
+        require(args.keytab != null, "Keytab must be specified when the keytab is specified")
```

Comment:
Looks like `SparkSubmitArguments` actually takes care of this and the initial login happens correctly even in the combination case (See https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala#L178 ).  Only the code in `Client.scala` needs to change to handle the combinations.